core/web/transfer_controller.go

```diff
@@ -40,9 +41,55 @@ func (tc *TransfersController) Create(c *gin.Context) {
 		return
 	}
 
-	db := tc.App.GetSqlxDB()
-	q := pg.NewQ(db, tc.App.GetLogger(), tc.App.GetConfig())
-	etx, err := bulletprooftxmanager.SendEther(q, chain.ID(), tr.FromAddress, tr.DestinationAddress, tr.Amount, chain.Config().EvmGasLimitTransfer())
+	if tr.FromAddress == utils.ZeroAddress {
+		jsonAPIError(c, http.StatusUnprocessableEntity, fmt.Errorf("invalid withdrawal source address address: %v", tr.FromAddress))
+		return
+	}
+
+	if !tr.AllowHigherAmounts {
+		balance := chain.BalanceMonitor().GetEthBalance(tr.FromAddress)
+
+		if balance == nil || balance.IsZero() {
+			jsonAPIError(c, http.StatusUnprocessableEntity, fmt.Errorf("balance is too low for this transaction to be executed: %v", balance))
+			return
+		}
+
+		var gasPrice *big.Int
+
+		gasLimit := chain.Config().EvmGasLimitTransfer()
+		estimator := chain.TxManager().GetGasEstimator()
+
+		if chain.Config().EvmEIP1559DynamicFees() {
+			gasPrice = chain.Config().EvmMaxGasPriceWei()
+			_, gasLimit, err = estimator.GetDynamicFee(gasLimit)
+			if err != nil {
+				jsonAPIError(c, http.StatusUnprocessableEntity, errors.Wrap(err, "failed to get dynamic gas fee"))
+				return
+			}
+		} else {
+			gasPrice, gasLimit, err = estimator.GetLegacyGas(nil, gasLimit)
+			if err != nil {
+				jsonAPIError(c, http.StatusUnprocessableEntity, errors.Wrap(err, "failed to estimate gas"))
+				return
+			}
+		}
+
+		tc.App.GetLogger().Criticalf("balance without fees: %v", balance)
+
+		intBalance := balance.ToInt()
+		fee := gasPrice.Mul(gasPrice, utils.NewBigI(int64(gasLimit)).ToInt())
+
+		intBalance = intBalance.Add(intBalance, fee)
+
+		tc.App.GetLogger().Criticalf("balance with fee: %v, expected to send: %v, result: %v", balance, tr.Amount.ToInt(), intBalance.Cmp(tr.Amount.ToInt()))
```

Comment:
Should this be using `Sub()` instead, to produce the actual result?
```suggestion
		tc.App.GetLogger().Criticalf("balance with fee: %v, expected to send: %v, result: %v", balance, tr.Amount.ToInt(), new(big.Int).Sub(intBalance, tr.Amount.ToInt()))
```

Reply:
Removed this log, was using it to debug something